tomb/src/pipelines/banyan_api/auth.rs

```diff
@@ -15,15 +21,29 @@ pub async fn pipeline(command: AuthSubCommand) -> Result<String> {
     let mut client = global.get_client().await?;
 
     // Process the command
-    let result = match command {
+    let result: Result<String, ClientError> = match command {
+        AuthSubCommand::RegisterDevice => {
+            // let device_key = EcEncryptionKey::generate().await?;
+            let private_device_key = GlobalConfig::from_disk().await?.api_key().await?;
+            // Register this device key
+            let credentials = register_device(client.clone(), private_device_key).await?;
+            // Update client authentication
+            client.with_credentials(credentials);
+            Ok("account registered".to_string())
+        }
+        #[cfg(feature = "fake")]
         AuthSubCommand::Register => {
+            // Additional imports
+            use tomb_common::banyan_api::requests::core::auth::fake_account::create::{
+                CreateAccountResponse, CreateFakeAccount,
+            };
             // Create local keys
             let api_key = EcSignatureKey::generate().await?;
             let public_api_key = api_key.public_key()?;
             let public_api_key_pem = String::from_utf8(public_api_key.export().await?)?;
             // Associate the key material with the backend
             let response: CreateAccountResponse = client
-                .call(CreateAccount {
+                .call(CreateFakeAccount {
                     device_api_key_pem: public_api_key_pem,
                 })
                 .await?;
```

Comment:
I don't think its supposed to be using the Fake account creation outside of tests right?

Reply:
true, this branch of the match statement only exists if you have the "fake" feature on- which is an easy way to ensure that you can still interact with the fake registration auth stuff through the CLI if you want to, but it doesn't work or present itself as an option in clap unless the feature is set